File: src/app/api/day5/route.ts
import { NextResponse } from 'next/server';

export const runtime = 'edge';

export async function POST(request: Request) {
  try {
    const { numbers } = await request.json() as { numbers: number[] };

    // 数値の配列を計算
    const arr = numbers;
    const mean = arr.reduce((a: number, b: number) => a + b, 0) / arr.length;
    const sorted = [...arr].sort((a: number, b: number) => a - b);
    const median = sorted.length % 2 === 0
      ? (sorted[sorted.length / 2 - 1] + sorted[sorted.length / 2]) / 2
      : sorted[Math.floor(sorted.length / 2)];
    const sum = arr.reduce((a: number, b: number) => a + b, 0);
    const std = Math.sqrt(
      arr.reduce((a: number, b: number) => a + Math.pow(b - mean, 2), 0) / arr.length
    );

    return NextResponse.json({
      result: {
        mean,
        median,
        std,
        sum
      }
    });
  } catch {
    return NextResponse.json({ error: 'リクエストの処理中にエラーが発生しました' }, { status: 500 });
  }
} 

File: src/app/api/day5/route.py
from http.server import BaseHTTPRequestHandler
import json
import numpy as np
from typing import Dict, Any

def calculate(numbers: list) -> Dict[str, float]:
    # 数値の配列をNumPy配列に変換
    arr = np.array(numbers)
    
    # 基本的な統計計算を実行
    result = {
        'mean': float(np.mean(arr)),
        'median': float(np.median(arr)),
        'std': float(np.std(arr)),
        'sum': float(np.sum(arr))
    }
    
    return result

def handler(request):
    # CORSヘッダーを設定
    headers = {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'POST, OPTIONS',
        'Access-Control-Allow-Headers': 'Content-Type',
        'Content-Type': 'application/json'
    }

    # OPTIONSリクエストの処理
    if request.method == 'OPTIONS':
        return {
            'statusCode': 200,
            'headers': headers,
            'body': ''
        }

    if request.method == 'POST':
        try:
            # リクエストボディを読み込み
            data = json.loads(request.body)
            
            # 計算を実行
            result = calculate(data['numbers'])
            
            return {
                'statusCode': 200,
                'headers': headers,
                'body': json.dumps({'result': result})
            }
        except Exception as e:
            return {
                'statusCode': 500,
                'headers': headers,
                'body': json.dumps({'error': str(e)})
            }
    
    return {
        'statusCode': 405,
        'headers': headers,
        'body': json.dumps({'error': 'Method not allowed'})
    } 